detection_rules/misc.py

```diff
@@ -32,6 +55,291 @@
+
+
+@dataclass
+class ReleaseManifest:
+
+    assets: Dict[str, AssetManifestMetadata]
+    assets_url: str
+    author: str  # author[login]
+    created_at: str
+    html_url: str
+    id: int
+    name: str
+    published_at: str
+    url: str
+    zipball_url: str
+    tag_name: str = None
+    description: str = None  # body
+
+
+class Manifest:
+    """Manifest handler for GitHub releases."""
+
+    def __init__(self, repo: str = 'elastic/detection-rules', release_name=None, tag_name=None, token=None):
+        self.repo_name = repo
+        self.release_name = release_name
+        self.tag_name = tag_name
+        self.gh_client = GithubClient(token)
+        self.has_token = token is not None
+
+        self.repo: Repository = self.gh_client.client.get_repo(repo)
+        self.release: GitRelease = get_gh_release(self.repo, release_name, tag_name)
+
+        if not self.release:
+            raise ValueError(f'No release found for {tag_name or release_name}')
+
+        if not self.release_name:
+            self.release_name = self.release.title
+
+        self.manifest_name = f'manifest-{self.release_name}.json'
+        self.assets: dict = self._get_enriched_assets_from_release()
+        self.release_manifest = self._create()
+        self.__release_manifest_dict = dataclasses.asdict(self.release_manifest)
+        self.manifest_size = len(json.dumps(self.__release_manifest_dict))
+
+    @property
+    def release_manifest_fl(self):
+        return io.BytesIO(json.dumps(self.__release_manifest_dict, sort_keys=True).encode('utf-8'))
+
+    def _create(self):
+        """Create the manifest from GitHub asset metadata and file contents."""
+        assets = {}
+        for asset_name, asset_data in self.assets.items():
+            entries = {}
+            data = asset_data['data']
+            metadata = asset_data['metadata']
+
+            for file_name, file_data in data.items():
+                file_metadata = file_data['metadata']
+
+                name = Path(file_name).name
+                file_metadata.update(name=name)
+
+                entry = AssetManifestEntry(**file_metadata)
+                entries[name] = entry
+
+            assets[asset_name] = AssetManifestMetadata(metadata['browser_download_url'], entries,
+                                                       metadata['zipped_sha256'], metadata['created_at'],
+                                                       metadata['label'])
+
+        release_metadata = self._parse_release_metadata()
+        release_metadata.update(assets=assets)
+        release_manifest = ReleaseManifest(**release_metadata)
+
+        return release_manifest
+
+    def _parse_release_metadata(self):
+        """Parse relevant info from GitHub metadata for release manifest."""
+        ignore = ['assets']
+        manual_set_keys = ['author', 'description']
+        keys = [f for f in list(ReleaseManifest.__annotations__) if str(f) not in ignore + manual_set_keys]
```

Comment:
dataclasses.fields I think is what you'd want

Reply:
```suggestion
        keys = [f.name for f in dataclasses.fields(ReleaseManifest) if f.name not in ignore + manual_set_keys]
```